### PR TITLE
[CI] Build cvxpy-base (Issue 1478)

### PR DESCRIPTION
**Pull request by phschiele**
Discussion: #1478 

@SteveDiamond as discussed, this additional action publishes `cvxpy-base` on PyPi, i.e. without any solvers preinstalled.

I have done a test run on [TestPyPI](https://test.pypi.org/project/cvxpy-base/), which worked well.
It can be tried using:

```
# Dependency not available on TestPyPI
pip install scipy
pip install --index-url https://test.pypi.org/simple/ cvxpy-base
```

It would only run with the next tagged release, but one could upload the current 1.1.15 version by disabling the `${{env.DEPLOY == 'True'}}` checks for one commit.

I have added a small section to the docs as well.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -101,3 +101,77 @@ jobs:
...
+
+      - name: Adapt setup.py
+        if: ${{env.DEPLOY == 'True'}}
+        shell: bash
+        run: |
+          # Mac has a different syntax for sed -i, this works across oses
+          sed -i.bak -e "s/name='cvxpy',/name='cvxpy-base',/g" setup.py
+          sed -i.bak '/osqp >= /d' setup.py
+          sed -i.bak '/ecos >= /d' setup.py
+          sed -i.bak '/scs >= /d' setup.py
```
Comment (phschiele) [inline .github/workflows/build.yml:149]: This is basically the complete logic and the assumptions made about the contents of `setup.py`.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -101,3 +101,77 @@ jobs:
...
+          rm -rf setup.py.bak
+
+      - name: Build wheels
+        if: ${{env.DEPLOY == 'True'}}
+        env:
+          CIBW_BUILD: "cp3${{env.PYTHON_SUBVERSION}}-*"
+          CIBW_SKIP: "*-win32 *-manylinux_i686"
+          CIBW_MANYLINUX_X86_64_IMAGE: manylinux_2_24
+        uses: joerick/cibuildwheel@v1.11.0
```
Comment (phschiele) [inline .github/workflows/build.yml:158]: Small technical detail: the wheels do not allow hyphens, so the wheel file is automatically renamed to `cvxpy_base*`. This has no effect on the pip installation.

File: .github/workflows/build.yml (hunk 1 above)
Comment (SteveDiamond) [reply .github/workflows/build.yml:149]: So basically are you removing the solvers from the dependencies?

File: doc/source/install/index.rst [hunk 3]
```diff
@@ -184,6 +184,27 @@ Install with SCIPY support
 CVXPY supports the SCIPY solver for LPs.
 This requires the `SciPy`_ package in Python which should already be installed as it is a requirement for CVXPY. `SciPy`_'s "interior-point" and "revised-simplex" implementations are written in python and are always available however the main advantage of this solver, is its ability to use the `HiGHS`_ LP solvers (which are written in C++) that comes bundled with `SciPy`_ version 1.6.1 and higher.
 
+Install without default solvers
+-------------------------
+
+CVXPY can also be installed without the default solver dependencies.
+This can be useful if the intention is to only use non-default solvers.
+
+The solver-less installation, ``cvxpy-base``, can currently be installed through pip and conda.
```
Comment (SteveDiamond) [inline doc/source/install/index.rst:193]: Are hyphenated package names normal in Python? I can't recall any.

File: .github/workflows/build.yml (hunk 1 above)
Comment (phschiele) [reply .github/workflows/build.yml:149]: Precisely 👍🏻 I thought it would be easier to maintain compared to having a second file.

**Review by SteveDiamond — APPROVED** [review]
LGTM

**Pull request merged**